### ISSUE TITLE
Ability to easily update Pygame and Pygame Gui and  make snakewm 'debuggable' within snakeware

### ISSUE THREAD

**Issue by Cyppa** [joshiemoore/snakeware#137]
As I am developing an app for snakeware I realised that my app was not compatible with the version of pygame gui that snakware uses. (My bad!) The app is several thousand lines of code already. If there was a way to update pygame and pygamegui in sankeware that would get this app into snakeware.

I have successfully built a snakeware ISO with the required pygame gui version (0.5.7) but snakewm won't load when I boot into the iso.

I am a noob when it comes to build root but if there is a way to debug snakewm within snakeware that would be a good first step.

Thanks.

**Comment by Admicos** [comment]
> If there was a way to update pygame and pygamegui in sankeware that
> would get this app into snakeware.

Pygame is included by upstream Buildroot, so Updating Buildroot will update
Pygame as well. For Pygame-Gui, the Buildroot package is at
snakeware/external/package/python-pygame-gui[1]. The .mk file contains the
metadata and download link, while the .hash file contains the md5 and sha256
hashes.

You can get all that info from https://pypi.org/pypi/pygame_gui/json, but you'll
need a JSON parser to deal with the data. If you use Firefox it automatically
detects JSON and provides a neat interface to search through it. Otherwise, try
using jq[2].

The data you're looking for is under releases["0.5.7"]

[1]: https://github.com/joshiemoore/snakeware/tree/master/snakeware/external/package/python-pygame-gui
[2]: https://stedolan.github.io/jq/

> I have successfully built a snakeware ISO with the required pygame
> gui version (0.5.7) but snakewm won't load when I boot into the iso.

This is probably something else. If you can send what you did to me (via
something like `git format-patch`), I could take a look, but I don't have a
snakeware dev env set up right now, so that'll take an hour or two.

> I am a noob when it comes to build root but if there is a way to
> debug snakewm within snakeware that would be a good first step.

A clean way to debug snakeware would definitely be nice. I'm not entirely sure
how that could be done, but maybe we could add a debug compile flag we can pass
to buildroot to make it redirect all Python text output to a file. (Or a serial
port to view via Qemu's virtual serial support)

**Comment by Cyppa** [comment]
Thanks for your help. I've been in touch with users named 'admicos' and 'Turtle1331' on Discord. Admicos showed me how to run python files using "os.system('sh')" command then running "python3 filename.py" to execute the app. This allowed me to open my app in snakeware and when it crashed it threw me back into a console. Real great way to help we move forwards.

Through this method I realised pygamegui 0.5.7 needs pygame 1.9.4 to run. This was becoming quite a task for me so I started re-writing parts of my code to work under pygamegui 0.5.6. This way I am able to test my app in snakeware with great result. This is just a temporary solution for now.

I hope to open a pull request and upload what I'm doing but I have no idea how to properly use github so that will take a while. As for biuldroot, that too is beyond me for now.

I am working on a text editor for snakeware and it is going great. I have a 'proof of concept' version ready with features that I disabled as I am re-writing the app to incorporate pygamegui 0.5.6 needs and window resizing. As I complete more of the re-writing I will release it somehow. For now I will upload it to the Discord chat under snakeware-dev. In that chat I  also uploaded videos of features that work under pygamegui 0.5.7 (without window resizing) if anyone want to see for themselves. 

Thanks agian for the help.
